### ISSUE TITLE
Some questions about Spatial Insight Engine

### ISSUE THREAD

**Issue by answer-miao** [M-3LAB/Look-Inside-for-More#8]
Thank you very much for your contribution in this field.

Because the train set of the Read3d-ad is complete, but the test set is half. So when extracting global features through projection, the features extracted during training and testing should be different. How can one answer this question theoretically?

Looking forward to your reply

**Comment by hzzzzzhappy** [comment]
Hi, 

This is a domain shift problem, meaning the training set and the test set differ significantly. In real industrial scenarios, we usually do not know in advance whether such a shift exists, so we do not apply any special handling.

If you want to forcibly align the training and test domains, you may refer to others’ work: [https://github.com/hustCYQ/GLFM-Multi-class-3DAD](https://github.com/hustCYQ/GLFM-Multi-class-3DAD), which splits half of the test set to match the training set.

Alternatively, you can refer to our test-time adaptation paper, which can automatically adjust the model to the fine-grained details of test samples: [https://github.com/M-3LAB/UCA-3DL](https://github.com/M-3LAB/UCA-3DL).

**Comment by answer-miao** [comment]
So what you mean is that although there are differences between the training set and the test set, after registration, this method is still effective despite its minor flaws. When I was debugging, by controlling the proportion during concat, it did have some effect.

**Comment by hzzzzzhappy** [comment]
I think so. However, during our experiments we didn't try to control the proportion, maybe you can try different proportion to valid this opinion.

**Comment by answer-miao** [comment]
Thank you for your reply. According to my recent experimental results, when concatenating pointMAE features, fpfh features, and global features, normalizing each of them before concatenating will yield better results. Thank you for your contribution！
